apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ClusterUtils.java

```diff
@@ -169,8 +169,11 @@ public static Set<ClusterInfo> evaluateClusterPath(File root, PropertyEvaluator
                     prj = _prj;
                 } else {
                     NbModuleProvider prov2 = _prj.getLookup().lookup(NbModuleProvider.class);
-                    if (prov2 != null && /* XXX is there a better way? */ cd.equals(prov2.getModuleJarLocation().getParentFile().getParentFile())) {
-                        prj = _prj;
+                    if (prov2 != null) {
+                        File fi = prov2.getModuleJarLocation();
+                        if (fi != null && /* XXX is there a better way? */ cd.equals(fi.getParentFile().getParentFile())) {
```

Comment:
`getParentFile().getParentFile()`
sounds like a second potential NPE ;)

Reply:
Technically yes; but no, not really. The contract is that a returned File is somewhere inside a project, it's parent can't be null.

For example, elsewhere in `ClusterUtils` there's
```
            File jar = nbmp.getModuleJarLocation();
            if (jar != null) {
                return jar.getParentFile().getParentFile();
            }
```
There are other examples scattered through other files.

Reply:
I hadn't seen this before
```
    /**
     * Returns location of built module JAR (file need not to exist).
     *
     * Currently (6.7) used only for suite-chaining. May return <tt>null</tt>,
     * module project cannot be chained into another suite in such case.
     * @return location of built module JAR
     */
    @CheckForNull File getModuleJarLocation();
```

Reply:
could be put into a utility method `getRootDir(NbModuleProvider prov)`. Would be self explanatory and could do the null check there.

Reply:
> could ... getRootDir(NbModuleProvider prov)

Something to consider for a rework of this ancient code, not holding my breath :-) Seems out of scope for fixing a missing null check.

Reply:
if you want you can add another commit. first fixes the bug, second extracts the two occurrences into a utility method. This keeps bugfix and code improvement separate.

There is no need to wait with small cleanups like this.